docs/en/operations/workload-scheduling.md

```diff
@@ -340,6 +340,46 @@ Blocked queries will wait indefinitely and not appear in `SHOW PROCESSLIST` unti
 
 Definitions of all workloads and resources in the form of `CREATE WORKLOAD` and `CREATE RESOURCE` queries are stored persistently either on disk at `workload_path` or in ZooKeeper at `workload_zookeeper_path`. ZooKeeper storage is recommended to achieve consistency between nodes. Alternatively `ON CLUSTER` clause could be used along with disk storage.
 
+## Configuration-based workloads and resources {#config_based_workloads}
+
+In addition to SQL-based definitions, workloads and resources can be predefined in the server configuration file. This is useful in cloud environments where some limitations are dictated by infrastructure, while other limits could be changed by customers. Configuration-based entities have priority over SQL-defined ones and cannot be modified or deleted using SQL commands.
+
+### Configuration format
+
+```xml
+<clickhouse>
+    <predefined_workloads>
+        <all>
+            <sql>WORKLOAD all SETTINGS max_io_requests = 1000 FOR s3disk_read, max_io_requests = 1000 FOR s3disk_write, max_bytes_per_second = 3355443200 FOR s3disk_read, max_bytes_per_second = 3355443200 FOR s3disk_write</sql>
+        </all>
+        <production>
+            <sql>WORKLOAD production IN all SETTINGS weight = 3</sql>
+        </production>
+    </predefined_workloads>
+    <predefined_resources>
+        <s3disk_read>
+            <sql>RESOURCE s3disk_read (READ DISK s3)</sql>
+        </s3disk_read>
+        <s3disk_write>
+            <sql>RESOURCE s3disk_write (WRITE DISK s3)</sql>
+        </s3disk_write>
+    </predefined_resources>
```

Comment:
Let's simplify this format further. We will just have <resources_and_workloads> with the plain text SQL queries inside. Queries are separated with a semicolon and may be a new line. This is the exact format that WorkloadEntityKeeperStorage uses. And to parse it, use the code similar to `WorkloadEntityKeeperStorage::refreshEntities()`. All queries in a section must be valid.